framework/test/Volo.Abp.FluentValidation.Tests/Volo/Abp/FluentValidation/ApplicationService_FluentValidation_Tests.cs

```diff
@@ -0,0 +1,167 @@
+                    MyStringValue3 = "ccc"
+                }
+            });
+
+            asyncOutput.ShouldBe("aaabbbccc");
+        }
+
+        [Fact]
+        public async Task Should_Not_Work_With_Wrong_Inputs()
+        {
+            // MyStringValue should be aaa, MyStringValue2 should be bbb. MyStringValue3 should be ccc
+
+            Assert.Throws<AbpValidationException>(() => _myAppService.MyMethod(new MyMethodInput
+            {
+                MyStringValue = "a",
+                MyMethodInput2 = new MyMethodInput2
+                {
+                    MyStringValue2 = "b"
+                },
+                MyMethodInput3 = new MyMethodInput3
+                {
+                    MyStringValue3 = "c"
+                }
+            }));
+
+            await Assert.ThrowsAsync<AbpValidationException>(async () => await _myAppService.MyMethodAsync(
+                new MyMethodInput
+                {
+                    MyStringValue = "a",
+                    MyMethodInput2 = new MyMethodInput2
+                    {
+                        MyStringValue2 = "b"
+                    },
+                    MyMethodInput3 = new MyMethodInput3
+                    {
+                        MyStringValue3 = "c"
+                    }
+                }));
+        }
+
+        [DependsOn(typeof(AbpAutofacModule))]
+        [DependsOn(typeof(AbpFluentValidationModule))]
+        public class TestModule : AbpModule
+        {
+            public override void PreConfigureServices(ServiceConfigurationContext context)
+            {
+                context.Services.OnRegistred(onServiceRegistredContext =>
+                {
+                    if (typeof(IMyAppService).IsAssignableFrom(onServiceRegistredContext.ImplementationType))
+                    {
+                        onServiceRegistredContext.Interceptors.TryAdd<FluentValidationInterceptor>();
+                    }
+                });
+            }
+
+            public override void ConfigureServices(ServiceConfigurationContext context)
+            {
+                context.Services.AddType<MyAppService>();
+            }
+        }
+
+        public interface IMyAppService
+        {
+            string MyMethod(MyMethodInput input);
+
+            Task<string> MyMethodAsync(MyMethodInput input);
+        }
+
+        public class MyAppService : IMyAppService, ITransientDependency
+        {
+            public string MyMethod(MyMethodInput input)
+            {
+                return input.MyStringValue + input.MyMethodInput2.MyStringValue2 + input.MyMethodInput3.MyStringValue3;
+            }
+
+            public Task<string> MyMethodAsync(MyMethodInput input)
+            {
+                return Task.FromResult(input.MyStringValue + input.MyMethodInput2.MyStringValue2 +
+                                       input.MyMethodInput3.MyStringValue3);
+            }
+        }
+
+        public class MyMethodInput : AbstractValidator<MyMethodInput>
```

Comment:
We may want to separate validators from the DTO. In this case, we should investigate validators on app startup and use them for validation.